frontend/src/metabase/hooks/use-sequenced-content-close-handler.ts

```diff
@@ -54,14 +54,41 @@ export function shouldClosePopover(
 
 export default function useSequencedContentCloseHandler() {
   const popoverDataRef = useRef<PopoverData>();
+  // A cheap hack to prevent popovers from being closed more than 1 time
+  // in a single sequence e.g. After users click there will be 2 events; mousedown -> click.
+  const hasPopoverClosedDuringMouseSequenceRef = useRef(false);
 
   const handleEvent = useCallback((e: MouseEvent | KeyboardEvent) => {
+    if (e instanceof MouseEvent) {
+      if (
+        popoverDataRef.current &&
+        shouldClosePopover(e, popoverDataRef.current) &&
+        !hasPopoverClosedDuringMouseSequenceRef.current
+      ) {
+        hasPopoverClosedDuringMouseSequenceRef.current = true;
+        popoverDataRef.current.close(e);
+      }
+    } else {
+      // keyboard event
+      if (
+        popoverDataRef.current &&
+        shouldClosePopover(e, popoverDataRef.current)
+      ) {
+        popoverDataRef.current.close(e);
+      }
+    }
+  }, []);
+
+  // Prevent toggling the popover after closing it with `mousedown`
+  const handleClickEvent = useCallback((e: MouseEvent) => {
```

Comment:
A new `handleClickEvent` to prevent closing and opening the popover when clicking on the trigger element.